### DISCUSSION TITLE
A few ideas / fixes

### DISCUSSION THREAD

**Discussion by Natooz** [SubmarinerApp/Submariner#126] [Ideas]
Hi 👋,

First thank you for your dedication to this project!
When using it I got a few ideas for enhancements / directions that I would like to suggest !

1. When adding a new server, if an error occurs, just show the error message (as opposed to closing the window, then trying to connect to the server and adding it in the list anyway)
2. Save configuration / sqlite files in a hidden cache directory (as opposed to ~/Music)
3. Offers an "album view" (like Spotify / iTunes) of libraries, that could be sorted by album or artist name
4. The app crashes when I try to download a file, then I can't access to the server anymore (the UI doesn't show anything, stays the same as the previous view)
5. When viewing albums, sometimes covers are not already downloaded, at first they are not showed, but you need to change view and come back so that it shown. There might be something to do here.

Also, I wondered if you had plans to extend the app to iOS / iPadOS / TVOS, using SwiftUI.

I am pretty busy right now and it's been ages since I touched swift / Xcode, but I'll be happy to help when I can.

**Comment by NattyNarwhal** [comment]
1. I think this is possible to refactor, but a bit annoying because connecting is done asynchronously.
2. I put it in ~/Music because it's easy to identify for backing up and cleaning up. I did lay the way for making this relocatable though (GH-121, plus stuff done a few months ago to avoid hardcoding paths in Core Data)
3. This would be nice. The server albums view, FWIW, is a thin shim around [getAlbumList](http://www.subsonic.org/pages/api.jsp#getAlbumList).
4. Do you see any messages in the console when it happens? Usually, that's a sign something hit an exception on the UI thread as it wedges all actions.
5. The cover stuff is a bit crumbly, yeah. 
6. I haven't yet. On iOS, I've been using play:Sub and Apple Music, so I haven't had the desire yet.
